src/org/labkey/test/components/domain/DomainFormPanel.java

```diff
@@ -139,21 +139,28 @@ else if (fieldDefinition.getType().equals(FieldDefinition.ColumnType.Time))
             else if (fieldDefinition.getType().equals(FieldDefinition.ColumnType.DateAndTime))
             {
 
-                // Identify the part of the format that is the date and the part that is the time. Take into account
-                // a data format may include spaces (time does not).
-                String format = fieldDefinition.getFormat().trim();
-                int index = format.lastIndexOf(" ");
+                // Identify the part of the format that is the date and the part that is the time.
```

Comment:
Ah, that explains it.  
This is probably out of scope for this fix, but this code (parsing date format/time format out of a format string) seems like it ought to be in a utils class like `TestDateUtils.java`